src/client.ts

```diff
@@ -27,13 +27,15 @@ export class Client implements ErdstallClient {
 	constructor(provider: ethers.providers.Provider | Signer, encConn: EnclaveReader | URL) {
 		this.provider = provider;
 		if(encConn! instanceof URL)
-			this.enclaveConn = Enclave.dial(encConn! as URL);
+			this.enclaveConn = Enclave.dial(encConn as URL);
 		else
-			this.enclaveConn = encConn! as EnclaveReader;
+			this.enclaveConn = encConn as EnclaveReader;
 		this.erdstallEventHandlerCache = new EventCache<ErdstallEvent>();
 		this.erdstallOneShotHandlerCache = new OneShotEventCache<ErdstallEvent>();
 	}
 
+	erdstall = this.erdstallConn!.erdstall;
```

Comment:
I think this executed only once, just before the constructor. A function would be better.